api/src/wfl/module/covid.clj

```diff
@@ -23,6 +24,61 @@
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
+(defn start-covid-workload!
+  [{:keys [submission-id workspace] :as workload}])
+
+(defn ^:private snapshot-error-map
+  "Return default ExceptionInfo map for snapshot-related exceptions."
+  [{:keys [executor items] :as _workload} {:keys [id snapshot_id] :as _record}]
+  {:workspace (:workspace executor)
+   :table items
+   :id id
+   :snapshot_id snapshot_id})
+
+(defn ^:private try-get-snapshot-reference
+  "Attempt to fetch snapshot reference with id SNAPSHOT_REFERENCE_ID
+  from EXECUTOR-derived workspace."
+  [{:keys [executor] :as workload} {:keys [snapshot_reference_id] :as record}]
+  (let [workspace (:workspace executor)]
+    (try
+      (rawls/get-snapshot-reference workspace snapshot_reference_id)
+      (catch Throwable cause
+        (throw (ex-info "Rawls unable to fetch snapshot reference"
+                        (util/deep-merge
+                         (snapshot-error-map workload record)
+                         {:snapshot_reference_id snapshot_reference_id})
+                        cause))))))
```

Comment:
Why catch and rethrow here?
Why `(assoc :snapshot_reference_id …)` here instead of in `snapshot-error-map`?

Reply:
I caught and rethrew because the original exception from Rawls API didn't share information about the request parameters (that I could see).

I also (try to) use this error map when we create a snapshot reference, if that fails we would not have a `:snapshot_reference_id` available.  But I think I can still put it in the map, since it should be in the underlying table schema.